clang/lib/AST/TypePrinter.cpp

```diff
@@ -2218,6 +2218,9 @@ printTo(raw_ostream &OS, ArrayRef<TA> Args, const PrintingPolicy &Policy,
     } else {
       if (!FirstArg)
         OS << Comma;
+      //if (Argument.getKind() == TemplateArgument::Type)
+      //  OS << "class ";
+
```

Comment:
TODO: if we go with this solution, this needs to happen here, but it should be under some conditions?

Reply:
I would move it into `printArgument`. If you have to, it would be find to add a new flag to PrintingPolicy. We have lots of fine-grained flags there.